### ISSUE TITLE
"Deprecated: Return type of ipl\Html\HtmlDocument::*" warnings with PHP 8.1

### ISSUE THREAD

**Issue by sbraz** [Icinga/icingaweb2#4938]
## Describe the bug
I see a lot of deprecation errors on the sidebar, such as:
```
Deprecated: Return type of ipl\Html\HtmlDocument::count() should either be compatible with Countable::count(): int, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /usr/share/icinga-php/ipl/vendor/ipl/html/src/HtmlDocument.php on line 434

Deprecated: Return type of ipl\Html\Attributes::offsetExists($name) should either be compatible with ArrayAccess::offsetExists(mixed $offset): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /usr/share/icinga-php/ipl/vendor/ipl/html/src/Attributes.php on line 456

Deprecated: Return type of ipl\Html\Attributes::offsetGet($name) should either be compatible with ArrayAccess::offsetGet(mixed $offset): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /usr/share/icinga-php/ipl/vendor/ipl/html/src/Attributes.php on line 472

Deprecated: Return type of ipl\Html\Attributes::offsetSet($name, $value) should either be compatible with ArrayAccess::offsetSet(mixed $offset, mixed $value): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /usr/share/icinga-php/ipl/vendor/ipl/html/src/Attributes.php on line 487

Deprecated: Return type of ipl\Html\Attributes::offsetUnset($name) should either be compatible with ArrayAccess::offsetUnset(mixed $offset): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /usr/share/icinga-php/ipl/vendor/ipl/html/src/Attributes.php on line 497

Deprecated: Return type of ipl\Html\Attributes::getIterator() should either be compatible with IteratorAggregate::getIterator(): Traversable, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /usr/share/icinga-php/ipl/vendor/ipl/html/src/Attributes.php on line 507
```

## To Reproduce

1. Switch from PHP 8.0 to PHP 8.1
2. Open Icinga Web 2

## Your Environment

* Icinga Web 2 version and modules (System - About): 2.11.1 (I can't open the System → About page to show more info because the warnings break the sidebar)
* Web browser used: Firefox
* Icinga 2 version used (`icinga2 --version`): r2.13.5-1
* PHP version used (`php --version`): 8.1.12
* Server operating system and version: Gentoo unstable

## Additional context
I found other bugs mentioning deprecation warnings for PHP 8.1 but I was under the impression that all fixes had landed in 2.11.0.

**Comment by sbraz** [comment]
Apologies, it looks like it was fixed in https://github.com/Icinga/ipl-html/pull/59 and Gentoo's icinga-php-library is still at v0.8.1 which didn't contain the fix.
@prometheanfire can you please bump icinga-php-library  to 0.10.0 to include the relevant fixes?

**Comment by lippserd** [comment]
> Apologies, it looks like it was fixed in [Icinga/ipl-html#59](https://github.com/Icinga/ipl-html/pull/59) and Gentoo's icinga-php-library is still at v0.8.1 which didn't contain the fix.
> @prometheanfire can you please bump icinga-php-library to 0.10.0 to include the relevant fixes?

No worries!
